### ISSUE TITLE
Rails error when eager loading and a count

### ISSUE THREAD

**Issue by zmoazeni** [mirakui/activerecord-mysql-index-hint#8]
Hi there, we have been using your library for a while. Thanks for maintaining it!

Today I ran into a curious Rails bug https://github.com/rails/rails/issues/24193 that stemmed from the way the hints are set. It's not specific to the hints, it has to do any queries that:

* Set `from(...)` to a string
* Use an `eager_load(:some_association)`
* Call `.count` (like most pagination libraries)

I'm pretty sure the problematic line is https://github.com/rails/rails/blob/4232f7ee427a510eb07e420f5883611bcb0abfae/activerecord/lib/active_record/relation/finder_methods.rb#L408 but we can avoid this by making this line truthy https://github.com/rails/rails/blob/4232f7ee427a510eb07e420f5883611bcb0abfae/activerecord/lib/active_record/relation/finder_methods.rb#L402

I came across another hinting library https://github.com/winebarrel/arel-mysql-index-hint but it is "sticky" in that once you set a hint for one query, all subsequent queries against the model will reuse that hint.

So I wanted to ping you before submitting a PR.

What I'm envisioning is something like this:

* We modify `Arel::Table` or create a subclass of `Arel::Table` that includes an index hint attribute
* We modify the Arel visitor to output the index (similar to https://github.com/winebarrel/arel-mysql-index-hint/blob/master/lib/arel-mysql-index-hint/arel-visitors-mysql.rb#L7)
* However where we deviate is in how we use this. The users of this library would still call the same method (`Product.where(user_id: 1).force_index(:idx_user_id).first`) but we would instantiate an Arel::Table instead of setting the string.

A rough sketch of the code would look like:

```ruby
def from_with_index_hint(hint_type, *args)
  return self if args.blank?
  indexes = args.map {|index| connection.quote_column_name index }
  temporary_arel_table = arel_table.dup # this is part of ActiveRecord
  temporary_arel_table.index_hint = "#{hint_type} INDEX(#{indexes.join(', ')})"
  self.from(temporary_arel_table)
end
```

So we just set the `from()` to a different `Arel::Table` and the overrided visitor that we write would output the hint.

What are your thoughts on a change like that?

**Comment by mirakui** [comment]
@zmoazeni Released #9 as v0.0.3. Thanks!

**Comment by zmoazeni** [comment]
Thank you!
